Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -143,51 +203,447 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
+bool KeyboardMouse::Point_Is_Inside_Octagon(const POINT& mouse_point, const POINT octagon_points[8])
+{
+  // This function uses the raycasting method to figure out if it's inside the octagon
+  // casting a ray horizontally in one direction from the mouse point and if the
+  // casted ray intersects with the edges of the octagon an even number of times (0 counts)
+  // the point is outside of the octagon. If the ray intersects an odd number of times, the
+  // ray is inside of the polygon. Jordan Curve Thereom is the name I was told
+  // https://erich.realtimerendering.com/ptinpoly/
+  unsigned int number_of_intersections = 0;
+
+  for (int i = 0; i < 8; i++)
+  {
+    int next_index = i + 1;
+    if (i == 7)
+      next_index = 0;
+
+    double rise = static_cast<double>(octagon_points[next_index].y) -
+                  static_cast<double>(octagon_points[i].y);
+    double run = static_cast<double>(octagon_points[next_index].x) -
+                 static_cast<double>(octagon_points[i].x);
+    double slope = rise / run;
+
+    // double temp_1 = mouse_point.y - octagon_points[i].y;
+    // double temp_2 = temp_1 / slope;
+    // double temp_3 = temp_2 + octagon_points[i].x;
+
+    double temp_x = ((mouse_point.y - octagon_points[i].y) / slope) + octagon_points[i].x;
+
+    if (temp_x < mouse_point.x)
+    {
+      continue;
+    }
+
+    if (i < 2 || i > 5)
+    {
+      if (temp_x >= octagon_points[i].x && temp_x < octagon_points[next_index].x)
+      {
+        number_of_intersections++;
+      }
+    }
+    else if (i >= 2 && i <= 5)
+    {
+      if (temp_x <= octagon_points[i].x && temp_x > octagon_points[next_index].x)
+      {
+        number_of_intersections++;
+      }
+    }
+  }
 
-  // Get the size of the current window (in my case Rect.top and Rect.left was zero).
-  RECT rect;
-  GetClientRect(s_hwnd, &rect);
+  if (number_of_intersections % 2 == 0)
+  {
+    return false;
+  }
+  else
+  {
+    return true;
+  }
+}
+
+double KeyboardMouse::Calculate_Distance_between_Points(const POINT& first_point,
+                                                        const POINT& second_point)
+{
+  // Pythagorean theoreom used to calculate the distance between points
+  // the hypotenuse of the right triangle is the distance.
+  // the x_difference and y_difference are the two legs of the triangle.
+  double x_difference = first_point.x - second_point.x;
+  double y_difference = first_point.y - second_point.y;
+
+  return sqrt((x_difference * x_difference) + (y_difference * y_difference));
+}
+
+long KeyboardMouse::Find_Second_Line_Point(const POINT& mouse_point, const POINT octagon_points[8],
+                                           long index_of_min_octagon_point)
+{
+  if (mouse_point.y < octagon_points[index_of_min_octagon_point].y)
+  {
+    switch (index_of_min_octagon_point)
+    {
+    case SOUTH:
+
+      break;
+    case SOUTH_EAST:
+
+      break;
+    case EAST:
+      return NORTH_EAST;
+      break;
```

Comment:
Consider removing `break` after a return, these can lead to dead code warnings and don't really do anything

Reply:
I prefer to leave the break in because I use the case and break as brackets when I use switch statements. I will take them out since they cause difficulties though.